nexus/reconfigurator/execution/src/dns.rs

```diff
@@ -742,6 +1043,549 @@ mod test {
+            cptestctx,
+            &opctx,
+            datastore,
+            &blueprint,
+            &overrides,
+            "squidport",
+            &dns_initial_internal,
+            &dns_initial_external,
+        )
+        .await;
+
+        // Now, go through the motions of provisioning a new Nexus zone.
+        // We do this directly with BlueprintBuilder to avoid the planner
+        // deciding to make other unrelated changes.
+        let sled_rows = datastore.sled_list_all_batched(&opctx).await.unwrap();
+        let zpool_rows =
+            datastore.zpool_list_all_external_batched(&opctx).await.unwrap();
+        let ip_pool_range_rows = {
+            let (authz_service_ip_pool, _) =
+                datastore.ip_pools_service_lookup(&opctx).await.unwrap();
+            datastore
+                .ip_pool_list_ranges_batched(&opctx, &authz_service_ip_pool)
+                .await
+                .unwrap()
+        };
+        let mut policy = policy_from_db(
+            &sled_rows,
+            &zpool_rows,
+            &ip_pool_range_rows,
+            // This is not used because we're not actually going through the
+            // planner.
+            NEXUS_REDUNDANCY,
+        )
+        .unwrap();
+        // We'll need another (fake) external IP for this new Nexus.
+        policy
+            .service_ip_pool_ranges
+            .push(IpRange::from(IpAddr::V4(Ipv4Addr::LOCALHOST)));
+        let mut builder = BlueprintBuilder::new_based_on(
+            &log,
+            &blueprint,
+            Generation::from(
+                u32::try_from(dns_initial_internal.generation).unwrap(),
+            ),
+            Generation::from(
+                u32::try_from(dns_latest_external.generation).unwrap(),
+            ),
+            &policy,
+            "test suite",
+        )
+        .unwrap();
+        let sled_id =
+            blueprint.sleds().next().expect("expected at least one sled");
+        let nalready = builder.sled_num_nexus_zones(sled_id);
+        let rv = builder
+            .sled_ensure_zone_multiple_nexus(sled_id, nalready + 1)
+            .unwrap();
+        assert_eq!(rv, EnsureMultiple::Added(1));
+        let blueprint2 = builder.build();
+        eprintln!("blueprint2: {:?}", blueprint2);
+        // Figure out the id of the new zone.
+        let zones_before = blueprint
+            .all_omicron_zones()
+            .filter_map(|(_, z)| z.zone_type.is_nexus().then_some(z.id))
+            .collect::<BTreeSet<_>>();
+        let zones_after = blueprint2
+            .all_omicron_zones()
+            .filter_map(|(_, z)| z.zone_type.is_nexus().then_some(z.id))
+            .collect::<BTreeSet<_>>();
+        let new_zones: Vec<_> = zones_after.difference(&zones_before).collect();
+        assert_eq!(new_zones.len(), 1);
+        let new_zone_id = *new_zones[0];
+
+        // Set this blueprint as the current target.  We set it to disabled
```

Comment:
Similar tiny nit as above, with apologies: "this blueprint" sounds like `blueprint2` (the one we just made), but first we have to set `blueprint` as the target. Could we set `blueprint` up above after we create, then keep the comment and just deal with `blueprint2` here?

Reply:
Done in 7028e0b92.